### DISCUSSION TITLE
Wrong version resolved after tycho update from 3.0.5 to 4.0.4

### DISCUSSION THREAD

**Discussion by marc-ernst** [eclipse-tycho/tycho#3349] [Q&A]
I'm building osgi plugins using bnd. With 3.0.5 they are resolved fine but with 4.0.4 it takes first the found plug-in from the target platform instead of the one I have build before. I'm using goal "bnd-process" and it's the first module listed in the pom. How can I prevent this without defining and specifying a fixed version number in the plug-in and feature?

**Comment by laeubi** [comment]
Can you share an example? You are maybe missing a dependency in the pom to your bnd-build item?

**Reply by marc-ernst** [reply]
No, I've added the dependencies in the feature pom.xml with exact version number. It looks like it searches first in the target platform for the dependency.

**Reply by laeubi** [reply]
The at best you should provide an [integration-test](https://github.com/eclipse/tycho/wiki#providing-an-integration-test) to demonstrate the issue, even though its a bit unusual to build something that's part of the target I think the reason is that pom dependencies are only checked after the target platform is queried but for reactor projects we should probably make an exception from that rule.

**Comment by marc-ernst** [comment]
[TYCHO3349bndfirst.zip](https://github.com/eclipse-tycho/tycho/files/13902215/TYCHO3349bndfirst.zip)

I have created a small reproducer. I'm not sure how to check the results automatically. If you look into the p2 repository after building with "mvn -Dtycho-version=4.0.4 install" you will find the plugin org.freemarker_2.3.22.v20160210-1233.jar instead of org.freemarker_2.3.32 .

**Reply by laeubi** [reply]
> BTW. tycho 4.0.4 has still this issue with the mvn proxy in settings. You need to remove the protocol and just specify one proxy to properly work with tycho 4.0.4. There was already an open issue but it was resolved because of this workaround. 

Have you tried 
try out the current [tycho snapshot 4.0.5](https://github.com/eclipse/tycho/wiki#getting-tycho-snapshots) build? If the proxy issue still persist please reopen the ticket with further information.

**Reply by marc-ernst** [reply]
--> even though its a bit unusual to build something that's part of the target
we do this because in mvn repository there might be newer versions of this plug-in which contains security/bug fixes.

**Reply by marc-ernst** [reply]
Running the same reproducer with tycho 3.0.5 will output the correct version org.freemarker_2.3.32 in the repository plug-in folder.

**Reply by marc-ernst** [reply]
Should I create an issue from this discussion?

**Reply by laeubi** [reply]
> Should I create an issue from this discussion?

Yes that seems a good next step.
